### DISCUSSION TITLE
Farbliche Hervorhebung der Eingabefelder beim DOI-Import

### DISCUSSION THREAD

**Discussion by vgerlach** [OPUS4/community#20] [General]
Derzeit werden beim DOI-Import befüllte Felder grün markiert. Dies ist gut so, allerdings geht diese Markierung verloren, wenn man weitere Metadaten (z.B. eine Sammlung) hinzufügt. Es wäre gut, wenn die Markierung erhalten bleibt.

Zudem gibt es den Vorschlag, folgende Farben zu verwenden:
        Feld rot, wenn benötigt und leer
        Feld grün, wenn ausgefüllt und identisch zu Daten des gewählten DOI (oder entsprechendes DOI item leer)
        Feld gelb, wenn ausgefüllt und verschieden zu Daten des gewählten DOI (Nutzer hat modifiziert)
        Feld weiß für fakultative Felder

**Comment by j3nsch** [comment]
Das ist in der aktuellen Umsetzung vielleicht etwas schwierig. Die Markierungen werden vom Javascript-Code beim befüllen der Felder gesetzt. Wenn die Formularstruktur verändert wird, also wenn z.B. für ein Sammlung, ein weiteres Formularfeld hinzugefügt wird, dann passiert das im Server und die Seite wird neu aufgebaut. Bereits im Formular vorhandene Daten bleiben dabei erhalten, aber der Code für die Markierung wird nicht erneut ausgeführt.

Ohne die Implementation jetzt grundlegend zu verändert, könnte man im Javascript ein Array pflegen, dass die automatisch befüllten Felder, auflistet. Wird der Eintrag in einem dieser Felder geändert, wird das Feld aus der Liste gestrichen und die Markierung entfernt. Das Array muss erhalten bleiben, wenn das Formular neu aufgebaut wird, muss also mit an den Server übergeben und dann wieder zurück übermittelt werden.

Die Farbe **gelb** für "automatische befüllt, aber dann geänderte Felder" scheint überflüssig. Ein Feld ist entweder befüllt oder leer. Es ist ein leeres Pflichtfeld oder nicht (rot). Es ist automatische befüllt worden oder nicht (grün). Warum eigentlich grün? Signalisiert das, dass die importierten Daten besonders vertrauenswürdig sind? Sollten ist natürlich hoffentlich sein, aber vielleicht integrieren wir ja später auch noch andere Quellen. :-)

FYI @alw-bsz
